processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/column/FixedWidthGroupByColumnSelectorStrategy.java

```diff
@@ -0,0 +1,214 @@
+    }
+  }
+
+  /**
+   * This is used for multi-valued dimensions, for values after the first one. None of the current types supported by
+   * this strategy handle multi-valued dimensions, therefore this short circuits and returns false
+   */
+  @Override
+  public boolean checkRowIndexAndAddValueToGroupingKey(
+      int keyBufferPosition,
+      Object rowObj,
+      int rowValIdx,
+      ByteBuffer keyBuffer
+  )
+  {
+    return false;
+  }
+
+  @Override
+  public int writeToKeyBuffer(
+      int keyBufferPosition,
+      ColumnValueSelector selector,
+      ByteBuffer keyBuffer
+  )
+  {
+    T value = getValue(selector);
+    int written = nullableTypeStrategy.write(keyBuffer, keyBufferPosition, value, keySizeBytes);
+    if (written < 0) {
+      throw DruidException.defensive("Unable to serialize the value [%s] to buffer", value);
+    }
+    // This strategy doesn't use dictionary building and doesn't hold any internal state, therefore size increase is nil.
+    return 0;
+  }
+
+  @Override
+  public Grouper.BufferComparator bufferComparator(
+      int keyBufferPosition,
+      @Nullable StringComparator stringComparator
+  )
+  {
+    return (lhsBuffer, rhsBuffer, lhsPosition, rhsPosition) -> {
+      T lhs = nullableTypeStrategy.read(lhsBuffer, lhsPosition + keyBufferPosition);
+      T rhs = nullableTypeStrategy.read(rhsBuffer, rhsPosition + keyBufferPosition);
+      if (stringComparator != null
+          && !DimensionComparisonUtils.isNaturalComparator(columnType.getType(), stringComparator)) {
+        return stringComparator.compare(String.valueOf(lhs), String.valueOf(rhs));
+      }
+      // Nulls are allowed while comparing
+      //noinspection ConstantConditions
+      return nullableTypeStrategy.compare(lhs, rhs);
+    };
+  }
+
+
+  @Override
+  public void reset()
+  {
+    // Nothing to reset
+  }
+
+  /**
+   * Returns true if the value at the selector is null. It unifies the null handling of primitive numeric types and the
+   * other types
+   */
+  private boolean selectorIsNull(ColumnValueSelector columnValueSelector)
+  {
+    if (isPrimitive && columnValueSelector.isNull()) {
+      return true;
+    }
+    return !isPrimitive && (columnValueSelector.getObject() == null);
+  }
+
+  /**
+   * Returns the value of the selector. It handles nullity of the value and casts it to the proper type so that the
+   * upstream callers donot need to worry about handling incorrect types (for example, if a double column value selector
+   * returns a long)
+   */
+  @Nullable
+  private T getValue(ColumnValueSelector columnValueSelector)
+  {
+    if (selectorIsNull(columnValueSelector)) {
+      return null;
+    }
+    // TODO(laksh): Check if calling .getObject() on primitive selectors be problematic??
+    // Convert the object to the desired type
+    //noinspection unchecked
+    return (T) DimensionHandlerUtils.convertObjectToType(columnValueSelector.getObject(), columnType);
```

Comment:
I think it makes sense given that either we'd be having numeric primitives, or complex (future work) types, since in that case, DimensionHandlerUtils.convertObjectToType() would be a no-op, and we can remove that call.